pkg/scorecard.go

```diff
@@ -80,11 +74,11 @@ func (r *RepoURL) Set(s string) error {
 }
 
 func RunScorecards(ctx context.Context, logger *zap.SugaredLogger,
-	repo RepoURL, checksToRun []checker.NamedCheck) <-chan Result {
-	resultsCh := make(chan Result)
+	repo RepoURL, checksToRun lib.CheckNameToFnMap) <-chan lib.CheckResult {
```

Comment:
Is there a specific reason to change slice into a custom type? `lib.CheckNameToFnMap`  `[]checker.NamedCheck`

go is about keeping it simple cognitive load. This would mean someone fixing/reading the code would have to go check type definition.

Reply:
The functionality we want is a map from string -> CheckFn. I don't agree that for the sake of readability we should choose a different data structure which increases code complexity (e.g we loop through all check names to find the fn instead of lookup).